MeesGame/Gameplay/GameObjects/TilesTypes/HoleTile.cs

```diff
@@ -2,7 +2,7 @@
 {
     class HoleTile : FloorTile
     {
-        public HoleTile(int layer = 0, string id = "") : base(TileType.Hole, layer, id)
+        public HoleTile(TileType tt = TileType.Hole, int layer = 0, string id = "") : base(tt, layer, id)
```

Comment:
Make a second, protected constructor that allows chaning the tile type. TileTypes may not be changed outside the tile classes.